src/elements/common/media-query/useMedia.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+
+import { toQuery, useMediaQuery } from 'react-responsive';
+import type { MediaFeatures } from './types';
+
+type MediaQuery = string | MediaFeatures;
+
+/**
+ * Formats the media query either as a MediaQuery object or string
+ * @param query
+ * @returns {string}
+ */
+function formatQuery(query: MediaQuery): string {
+    return typeof query === 'string' ? query : toQuery(query);
+}
+
+/**
+ * Executes media query
+ * @param query
+ * @param onQueryChange
+ * @returns {boolean}
+ */
+function useQuery(query: string, onQueryChange?: (_: boolean) => void): boolean {
+    return useMediaQuery({ query }, null, onQueryChange);
+}
+
+/**
+ * @param query
+ * @param onQueryChange
+ * @returns {boolean}
+ */
+function useMedia(query: string, onQueryChange?: (_: boolean) => void): boolean {
```

Comment:
I think we should consider the hook the primary API, rather than the HOC. So any utility logic that we would put into the HOC to simplify Box-specific responsive usage should go here, too.

Reply:
useMedia now returns a media props object